crates/fuel-core/src/database.rs

```diff
@@ -520,18 +493,486 @@ mod tests {
         }
     }
 
-    #[test]
-    fn column_keys_not_exceed_count_test_on_chain() {
-        column_keys_not_exceed_count::<OnChain>();
+    mod on_chain {
+        use super::*;
+        use crate::database::{
+            database_description::on_chain::OnChain,
+            DatabaseHeight,
+        };
+        use fuel_core_storage::{
+            tables::Coins,
+            transactional::WriteTransaction,
+        };
+        use fuel_core_types::blockchain::block::CompressedBlock;
+
+        #[test]
+        fn column_keys_not_exceed_count_test() {
+            column_keys_not_exceed_count::<OnChain>();
+        }
+
+        #[test]
+        fn database_advances_with_a_new_block() {
+            // Given
+            let mut database = Database::<OnChain>::default();
+            assert_eq!(database.latest_height().unwrap(), None);
+
+            // When
+            let advanced_height = 1.into();
+            database
+                .storage_as_mut::<FuelBlocks>()
+                .insert(&advanced_height, &CompressedBlock::default())
+                .unwrap();
+
+            // Then
+            assert_eq!(database.latest_height().unwrap(), Some(advanced_height));
+        }
+
+        #[test]
+        fn database_not_advances_without_block() {
+            // Given
+            let mut database = Database::<OnChain>::default();
+            assert_eq!(database.latest_height().unwrap(), None);
+
+            // When
+            database
+                .storage_as_mut::<Coins>()
+                .insert(&Default::default(), &Default::default())
```

Comment:
It's not quite clear what the `without_block` is here. `Default` obscures what is actually being inserted.